SPECS/tdnf/tdnf-add-installonlypkgs-config.patch

```diff
@@ -0,0 +1,25 @@
+From de620a8166329c2f7b6eaecb5b4e24af1db95e4d Mon Sep 17 00:00:00 2001
+From: Sam Meluch <sam.meluch@microsoft.com>
+Date: Tue, 19 Dec 2023 12:04:36 -0800
+Subject: [PATCH] Add installonlypkgs to tdnf.conf for mariner kernel packages
+
+---
+ etc/tdnf/tdnf.conf | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/etc/tdnf/tdnf.conf b/etc/tdnf/tdnf.conf
+index 67c4d51..8ec4003 100644
+--- a/etc/tdnf/tdnf.conf
++++ b/etc/tdnf/tdnf.conf
+@@ -1,7 +1,8 @@
+ [main]
+ gpgcheck=1
+ installonly_limit=3
++installonlypkgs=kernel kernel-hci kernel-mos kernel-mshv kernel-uvm kernel-uvm-cvm kernel-azure
+ clean_requirements_on_remove=0
```

Comment:
do we want this off by default?  I think it's enabled by default for DNF, per [here](https://www.man7.org/linux/man-pages/man8/yum2dnf.8.html):
The clean_requirements_on_remove switch is on by default in DNF.

Reply:
I guess it would be good to stay consistent with how tdnf has been working, regardless of going with yum vs. dnf.

Reply:
By default this is off in tdnf and we have not changed it, see [here](https://github.com/vmware/tdnf/blob/dev/etc/tdnf/tdnf.conf#L4) for the upstream file. This would be a reasonable change in behavior that we may not want for 2.0. We may be able to investigate for 3.0 as it could potentially cause some splash by turning it on now.